spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/async/LazyTraceScheduledThreadPoolExecutorTests.java

```diff
@@ -37,12 +100,532 @@ protected void finalize() {
 				wasCalled.set(true);
 			}
 		};
-		BeanFactory beanFactory = BDDMockito.mock(BeanFactory.class);
+		BeanFactory beanFactory = mock(BeanFactory.class);
 
 		new LazyTraceScheduledThreadPoolExecutor(10, beanFactory, executor, null).finalize();
 
 		BDDAssertions.then(wasCalled).isFalse();
 		BDDAssertions.then(executor.isShutdown()).isFalse();
 	}
 
+	@Test
```

Comment:
Can we separate `given`, `when` and `then` sections? I can't figure out which line is responsible for calling the actual code to test. I guess it's this one `assertThat((Future<?>) executor.decorateTask(runnable, value)).isEqualTo(future); ` ?

Reply:
Yep, I'll go ahead and do that. In the example you cite, that statement is both the when and then. It's verifying that the return value for that method matches the delegate's return value. That's not super clear, so let me rework those a bit

Reply:
Reworked the tests a bit. Lemme know what you think